cpp/daal/src/algorithms/tsne/tsne_gradient_descent_fpt.cpp

```diff
@@ -691,24 +704,24 @@ services::Status integrationKernelImpl(const DataType eta, const DataType moment
         DataType * localSum = sumTlsData.local(tid);
         for (IdxType i = iStart; i < iEnd; ++i)
         {
-            const DataType dx = exaggeration * attrx[i] - zNorm * repx[i];
-            const DataType dy = exaggeration * attry[i] - zNorm * repy[i];
+            const DataType dx = exaggeration * attrX[i] - zNorm * repX[i];
+            const DataType dy = exaggeration * attrY[i] - zNorm * repY[i];
             localSum[0] += dx * dx + dy * dy;
 
-            gx = (dx * (ux = oldForcex[i]) < DataType(0)) ? gainx[i] + 0.2 : gainx[i] * 0.8;
+            gx = (dx * (ux = oldForceX[i]) < DataType(0)) ? gainX[i] + 0.2 : gainX[i] * 0.8;
             if (gx < 0.01) gx = 0.01;
```

Comment:
What a magical numbers?

Reply:
changing `gain` aligned with scikit-learn and cuml.

Reply:
What about using `constexpr` for such constant decalration?